src/test/java/de/zalando/aruha/nakadi/validation/EventBodyMustRespectSchema.java

```diff
@@ -0,0 +1,233 @@
+        return effectiveSchema;
+    }
+
+    public void setEffectiveSchema(final JSONObject effectiveSchema) {
+        this.effectiveSchema = effectiveSchema;
+    }
+}
+
+class JSONSchemaValidator implements EventValidator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JSONSchemaValidator.class);
+
+    private final Schema schema;
+    private final JSONObject effectiveSchema;
+
+    public JSONSchemaValidator(final JSONObject effectiveSchema) {
+        this.effectiveSchema = effectiveSchema;
+        schema = SchemaLoader.load(effectiveSchema);
+    }
+
+    @Override
+    public boolean accepts(final JSONObject event) {
+        try {
+            schema.validate(event);
+            return true;
+        } catch (final ValidationException e) {
+            e.printStackTrace();
+            LOG.error("Validation failed: {}. Schema was: {}",
+                new Object[] {Joiner.on(";").join(e.getCausingExceptions()), effectiveSchema, e});
+            return false;
+        }
+    }
+}
+
+class QualifiedJSONSchemaValidator implements EventValidator {
+    private Qualifier qualifier;
+    private JSONSchemaValidator validator;
+
+    public QualifiedJSONSchemaValidator(final Qualifier qualifier, final JSONSchemaValidator jsv) {
+        this.qualifier = qualifier;
+        this.validator = jsv;
+    }
+
+    public Qualifier getQualifier() {
+        return qualifier;
+    }
+
+    public void setQualifier(final Qualifier qualifier) {
+        this.qualifier = qualifier;
+    }
+
+    public JSONSchemaValidator getValidator() {
+        return validator;
+    }
+
+    public void setValidator(final JSONSchemaValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public boolean accepts(final JSONObject event) {
+
+        return validator.accepts(event);
+    }
+}
+
+class QualifiedJSONSchemaValidationChain implements EventValidator {
+    private final List<QualifiedJSONSchemaValidator> qualifiedValidators;
+    private final JSONSchemaValidator defaultSchemaValidator;
+
+    public QualifiedJSONSchemaValidationChain(final List<QualifiedJSONSchemaValidator> qualifiedValidators,
+            final JSONSchemaValidator defaultSchemaValidator) {
+        this.qualifiedValidators = qualifiedValidators;
+        this.defaultSchemaValidator = defaultSchemaValidator;
+    }
+
+    @Override
+    public boolean accepts(final JSONObject event) {
+        final EventValidator validator = qualifiedValidators.stream().filter(it ->
+                                                                    it.getQualifier().getMatch().equals(
+                                                                        event.getString(it.getQualifier().getField())))
+                                                            .map(EventValidator.class::cast).findFirst().orElse(
+                                                                defaultSchemaValidator);
```

Comment:
We should find some better solution here. But this is a nice peace of code to check the formatters if we decide to have a look on some other possibilities like `CheckStyle` :D